pytube/request.py

```diff
@@ -86,6 +135,56 @@ def filesize(url: str) -> int:
     return int(head(url)["content-length"])
 
 
+@lru_cache(maxsize=None)
```

Comment:
definitely should bound here, especially if this is used on low-memory devices (e.g. raspberry pi's)

Reply:
I can definitely change this to not have a maxsize argument (default value is 128), but I think it makes more sense to just cache it. Storing an integer in cache is probably going to be preferable to making a network request, especially because of YouTube's rate-limiting

Reply:
😄  tends to be good practice to set /some/ sensibile limit.

Reply:
I removed the maxsize argument, so now it will only cache up to 128